task-list/app/views/tasks/index.html.erb

```diff
@@ -2,8 +2,25 @@
 <h1>To Do:</h1>
 <ul>
   <% @tasks.each do |task| %>
-    <li><%= link_to task.name, "/tasks/#{task.id}" %>  <%= link_to "[x]", "/tasks/#{task.id}", method: :delete, data: { confirm: 'Confirm Delete' }, class: "delete-link" %></li>
+    <%if task.completed_at == nil %>
+      <li>
+        <%= link_to '<i class="fa fa-square-o fa-fw"></i>'.html_safe, "tasks/#{task.id}/complete", method: :patch %>
+        <%= link_to task.name, "/tasks/#{task.id}" %>
+        <%= link_to '<i class="fa fa-pencil"></i>'.html_safe, "/tasks/#{task.id}/edit", class: "edit-link" %>
+        <%= link_to '<i class="fa fa-trash-o"></i>'.html_safe, "/tasks/#{task.id}", method: :delete, data: { confirm: 'Confirm Delete' }, class: "delete-link" %>
+      </li>
+    <% end %>
+  <% end %>
+  <% @tasks.each do |task| %>
```

Comment:
I like that you're looping through `@tasks` twice to get each of the incomplete and completed tasks into separate groups. However, because there's so much repetition in the code within each loop, I would recommend maybe putting those bits into a partial.